### PR TITLE
Convert the repository into an all-purpose flake collection

### PR DESCRIPTION
**Pull request by chris-martin**
Following up on discussion with Pat on Friday. The plan is to then rename this repo to `flakes`.

### CODE REVIEW

File: main/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Packages should generally be aliased in several ways so that each use site
+can choose what level of specificity in the fourmolu version it wants.
+For example, we have:
+
+- `fourmolu.default` -
+  If you don't care what version of fourmolu you get
+- `fourmolu.version-0-13-x` -
+  If you want major version 0.13 but don't care what minor version
+- `fourmolu.version-0-13-0` -
```
Comment (pbrisbin) [inline main/readme.md:18]: Not that I really care, but more just curious: is there a limitation that prevents these from being named `fourmolu`, `fourmolu.0.13` and `fourmolu.0.13.0`?

**Review by pbrisbin — APPROVED** [review]
Cool!

File: main/readme.md (hunk 1 above)
Comment (chris-martin) [reply main/readme.md:18]: The nix language does let you put `.` (or anything at all) in attribute keys as long at you write them in quotes, but I'm not sure to what extent the command-line interface can deal with it. I think at the very least dealing with them would be inconvenient.

**Review by stackptr — APPROVED** [review]
Love it

File: main/readme.md (hunk 1 above)
Comment (chris-martin) [reply main/readme.md:18]: I'm not sure what conventions apply. Nixpkgs generally doesn't keep multiple versions of a package around and so it doesn't do this at all. When they do, it's pretty inconsistent. The way they do GHC is horrifying e.g. expressing 9.2.7 as "ghc927". My habit is to err on the side of not abbreviating things, but happy to collapse "version" to "v" as that's pretty standard.

**Pull request merged**